crates/http/src/lib.rs

```diff
@@ -59,15 +59,15 @@ pub struct HttpTrigger {
 #[derive(Args)]
 pub struct CliArgs {
     /// IP address and port to listen on
-    #[clap(long = "listen", default_value = "127.0.0.1:3000", value_parser = parse_listen_addr)]
+    #[arg(long = "listen", default_value_t = std::net::SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(127, 0, 0, 1), 3000)), value_parser = parse_listen_addr)]
```

Comment:
Q: Can you elaborate on this update (use of `default_value_t`)?  Is it utilizing a performance or efficiency gain w/ clap 4.x.x?

Reply:
I think this possibly relates to correctness more than efficiency - by writing the _actual constructor call_ rather than a string, the compiler can check that the default value is well formed and of the correct type.

Reply:
I think I was going somewhere with that. Ended up leaving it due to the reasons @itowlson mentioned. Also seems to be preferred when reading clap's docs.